packages/admin-next/dashboard/src/v2-routes/campaigns/common/components/create-campaign-form-fields/create-campaign-form-fields.tsx

```diff
@@ -0,0 +1,268 @@
+              </Form.Item>
+            )
+          }}
+        />
+
+        <Form.Field
+          control={form.control}
+          name={`${fieldScope}ends_at`}
+          render={({ field: { value, onChange, ref: _ref, ...field } }) => {
+            return (
+              <Form.Item>
+                <Form.Label>{t("campaigns.fields.end_date")}</Form.Label>
+
+                <Form.Control>
+                  <DatePicker
+                    showTimePicker
+                    value={value ?? undefined}
+                    onChange={(v) => onChange(v ?? null)}
+                    {...field}
+                  />
+                </Form.Control>
+
+                <Form.ErrorMessage />
+              </Form.Item>
+            )
+          }}
+        />
+      </div>
+
+      <div>
+        <Heading>{t("campaigns.budget.create.header")}</Heading>
+        <Text size="small" className="text-ui-fg-subtle">
+          {t("campaigns.budget.create.hint")}
+        </Text>
+      </div>
+
+      <Form.Field
+        control={form.control}
+        name={`${fieldScope}budget.type`}
+        render={({ field }) => {
+          return (
+            <Form.Item>
+              <Form.Label>{t("campaigns.budget.fields.type")}</Form.Label>
+
+              <Form.Control>
+                <RadioGroup
+                  className="flex gap-y-3"
+                  {...field}
+                  onValueChange={field.onChange}
+                >
+                  <RadioGroup.ChoiceBox
+                    className={clx("basis-1/2", {
+                      "border-2 border-ui-border-interactive":
+                        "spend" === field.value,
+                    })}
+                    value={"spend"}
+                    label={t("campaigns.budget.type.spend.title")}
+                    description={t("campaigns.budget.type.spend.description")}
+                  />
+
+                  <RadioGroup.ChoiceBox
+                    className={clx("basis-1/2", {
+                      "border-2 border-ui-border-interactive":
+                        "usage" === field.value,
+                    })}
+                    value={"usage"}
+                    label={t("campaigns.budget.type.usage.title")}
+                    description={t("campaigns.budget.type.usage.description")}
+                  />
+                </RadioGroup>
+              </Form.Control>
+              <Form.ErrorMessage />
+            </Form.Item>
+          )
+        }}
+      />
+
+      <div className="grid grid-cols-1 gap-4 md:grid-cols-2">
+        <Form.Field
+          control={form.control}
+          name={`${fieldScope}budget.limit`}
+          render={({ field: { onChange, value, ...field } }) => {
+            return (
+              <Form.Item className="basis-1/2">
+                <Form.Label>{t("campaigns.budget.fields.limit")}</Form.Label>
+
+                <Form.Control>
+                  {isTypeSpend ? (
+                    <CurrencyInput
```

Comment:
suggestion: this should be disabled until currency is selected so we can display proper currency input (we usually add an info tooltip while on FormLabel in such cases)

Reply:
I tried this, but disabled doesn't seem to be working with currency input

Reply:
this ` disabled={!currencyValue}` should work if you pass after props spread `{...props}` since it's probably overriding the prop

Reply:
in the new branch as well